src/engraving/rendering/dev/systemlayout.cpp

```diff
@@ -2404,30 +2405,50 @@ double SystemLayout::minVertSpaceForCrossStaffBeams(System* system, staff_idx_t
                 if (!beam || !beam->cross() || !beam->autoplace() || beam->elements().front() != item) {
```

Comment:
A small suggestion for efficiency - as we now cache the beams cross staff position (above/below/between), this can be checked here 
`... || beam->ldata()->crossStaffBeamPos != BeamBase::CrossStaffBeamPosition::BETWEEN)`